R/pkg/tests/fulltests/test_mllib_classification.R

```diff
@@ -307,13 +307,13 @@ test_that("spark.mlp", {
   expect_equal(summary$numOfOutputs, 3)
   expect_equal(summary$layers, c(4, 5, 4, 3))
   expect_equal(length(summary$weights), 64)
-  expect_equal(head(summary$weights, 5), list(-0.878743, 0.2154151, -1.16304, -0.6583214, 1.009825),
+  expect_equal(head(summary$weights, 5), list(0.327309, 0.2385232, -0.8763775, -1.01558, 0.8494107),
                tolerance = 1e-6)
 
   # Test predict method
   mlpTestDF <- df
   mlpPredictions <- collect(select(predict(model, mlpTestDF), "prediction"))
-  expect_equal(head(mlpPredictions$prediction, 6), c("0.0", "1.0", "1.0", "1.0", "1.0", "1.0"))
+  expect_equal(head(mlpPredictions$prediction, 6), c("2.0", "2.0", "2.0", "2.0", "2.0", "2.0"))
```

Comment:
This change concerns me; those predictions are all wrong now according to the data. It probably means the test was insufficient to begin with. I think the tolerance parameter is way too high; unset it if possible or use a much smaller value like 0.00001

Reply:
I tried to set `tol = 0.00001`:
```
> head(summary$weights, 5)
[[1]]
[1] -24.28415

[[2]]
[1] 107.8701

[[3]]
[1] 16.86376

[[4]]
[1] 1.103736

[[5]]
[1] 9.244488

> mlpTestDF <- df
>   mlpPredictions <- collect(select(predict(model, mlpTestDF), "prediction"))
> head(mlpPredictions$prediction, 6)
[1] "1.0" "1.0" "1.0" "1.0" "0.0" "1.0"
```

Reply:
If it is ok, I will commit this.

Reply:
Much better -- those are actually the correct answers!